### ISSUE TITLE
Touchscreen issue with End Turn

### ISSUE THREAD

**Issue by limburgher** [wesnoth/wesnoth#9684]
### Game and System Information

- **Version:** 1.18.2 and 1.19.7
- **OS:** Fedora 41

### Description of the bug

End Turn button not working properly on a touch screen

### Steps to reproduce the behavior

1.18.2
start the game
start tutorial
try to end your turn when asked.
Actual results:
button does not react.
Expected results:
working buttons.

### Additional context

1.19.7:
I open the tutorial, 
click the messages,
make me moves the tutorial requires 
but i can't press "end turn" UNTIL i moved the mouse there and than i can use it.

After the mouse movement fixed that issue, it was no issue any longer for the rest of the tutorial.

10:1 it's an event handling problem, because the touchscreen does not produce correct (key,button)-UP Events or in a incorrect order for the normal usage via mouse. I have seen this many times, touchscreens have been those issues i.e. in webpages to drag&drop elements.

**Comment by Wedge009** [comment]
@babaissarkar Just wondering: is Android's handling of touch events separate from how Wesnoth generally interprets inputs like mouse clicks?

**Comment by babaissarkar** [comment]
> [@babaissarkar](https://github.com/babaissarkar) Just wondering: is Android's handling of touch events separate from how Wesnoth generally interprets inputs like mouse clicks?

Android isn't involved. It's always Wesnoth itself handling keys. For double tap etc there's some minor `#ifdef __ANDROID__` involved. Touch gets transparently mapped to clicks with a bit of extra info that's it's touch.

**Comment by Wedge009** [comment]
I suppose there's an underlying inaccurate assumption, then, that only Android builds have inputs via touch-screen. I imagine all-in-one devices are probably the most common form factor with a touch-screen, for a desktop context.

**Comment by babaissarkar** [comment]
> I suppose there's an underlying inaccurate assumption, then, that only Android builds have inputs via touch-screen. I imagine all-in-one devices are probably the most common form factor with a touch-screen, for a desktop context.

Yes, a desktop with touchscreen support or iOS devices also use same code unless it's ifdef guarded. In fact there was an issue for a global touchscreen device.
